mcp/client.go

```diff
@@ -127,6 +131,12 @@ type ClientSessionOptions struct {
 }
 
 func (c *Client) capabilities() *ClientCapabilities {
+	// If Capabilities is set, use it as the override.
+	if c.opts.Capabilities != nil {
+		return c.opts.Capabilities
+	}
+
+	// Otherwise, infer capabilities from handlers.
```

Comment:
it feels a little foot gunny for the `Capabilities` struct to, e.g., disable a feature when the user also has a handler registered for it. should we at least emit warnings when they do this?

Reply:
Ack, let's review in a separate PR.